packages/frontend/src/creation-form/components/approve-reward.tsx

```diff
@@ -49,11 +50,18 @@ export const ApproveReward = ({
         if (!approveAsync) return;
         let cancelled = false;
         const approve = async () => {
-            const tx = await approveAsync();
-            const receipt = await publicClient.waitForTransactionReceipt({
-                hash: tx.hash,
-            });
-            if (!cancelled) onApprove(receipt);
+            setLoading(true);
```

Comment:
Initially I assumed the `isLoading` thing in the `useContractWrite` hook would be set to true while the tx is being sent, but I was wrong apparently. I'm wondering if it makes sense to rename the `approving` var now. The `isLoading` is apparently triggered while the tx is presented to the user to the wallet and set to false when the tx is sent to the blockchain.

Reply:
Fixed in 334b99284497c3e9f29120f41c5258d91fca98be.